mm2src/lp_ordermatch.rs

```diff
@@ -3368,7 +3381,9 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
         .with_price(req.price)
         .with_conf_settings(conf_settings);
 
-    let new_order = try_s!(builder.build());
+    let mut new_order = try_s!(builder.build());
+    new_order.save_in_history = req.save_in_history;
```

Comment:
This might be acceptable as a temporary solution in some situations e.g. we need big refactoring to avoid `let mut`. But we already have a builder for MakerOrder, please use it too for `save_in_history`.